stdlib/README.adoc

```diff
@@ -0,0 +1,102 @@
+= OCaml Stdlib
+
+== Design principles
+
+* When a module `M` has a main type it must be named `M.t`.
+* Functions should take the main type `t` as first argument (unless
+  there is a good reason for it, see next point).
```

Comment:
I'm not sure that this is established, either. I wish there was a definitive order, but there doesn't seem to be, and the pros and cons should be thought through.

Reply:
Exactly.  That's the whole point of the discussion I would like we have here!

Reply:
We will end up mixing conversations on many non-consensual points.  Let's create a more high-level document, and suggest specific additions for it, so that they can be discussed independently.

Reply:
I merged your input at the beginning of the document making clear that the rest are points to discuss.  Or would you like to split the discussion somehow?

Reply:
I think it's good to settle on some points where we don't currently have consensus. (NB: In English, "non-consensual" means "without consent", and not "without consensus". It has very bad connotations, usually involving violence and such. You probably should say instead "we'll end up mixing conversations on many points where no common consensus exists" or some such.)

Reply:
The current PR proposes too many changes compared to the current design.  It's good to discuss possible evolution of stdlib principles, but as I wrote above, I think each of those evolution should be proposed and discussed independently.  For instance, about the addition of labels on functional argument: why not (even though I'm not personally convinced), but this is not the current style, and it would not make sense to enforce this style on new additions without thinking on how to evolve the current stdlib.

Reply:
(Thanks @pmetzger for the remark about  consensus vs consensual.)

Reply:
Maybe it would be good to establish some sort of mechanism for bringing up and independently discussing each of the issues in question? Unfortunately neither Mantis nor Github's PR mechanism is well suited to it, but it would be nice to have some way to make progress on going over the full spectrum of issues.

Reply:
GitHub PRs to suggest and discuss additions to a "planned evolutions" section might work.  My point is that one cannot discuss all evolutions at once; each of them deserves its own discussion (including on the backward compatibility story).